docs/xdebug.md

```diff
@@ -0,0 +1,68 @@
+# Debugging PHP in Archipelago
+
+This document describes how to enable Xdebug for local PHP development using the PHPStorm IDE and a docker container running the Archipelago `esmero-php:development` image. It involves interacting with the [esmero/archipelago-docker-images](https://github.com/esmero/archipelago-docker-images) repo and the [esmero/archipelago-deployment](https://github.com/esmero/archipelago-deployment) repo.
+
+## Part 1: Docker
+1. Find out what your local machine (your actual computer!) IP address is. For example mine is, `192.168.1.3`
+
+1. Build the development image with Xdebug enabled. Run the following from the `/archipelago-docker-images/esmero-php-fmp` folder.
+`sudo docker build -f Dockerfile.dev --build-arg LOCAL_IP=<your machine IP> -t esmero-php:development . --no-cache`
+
+2. Run the following commands from your `/archipelago-deployment` directory:
+   
+   `docker-compose down` \
+   `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d`
+   
+   This version of `docker-compose up` uses an override file and to specify we want to use the development image for our PHP service. It will rebuild the containers and now you are ready for Part 2.
+
+## Part 2: PHPStorm
+
+1. In PHPStorm, open your `archipelago-deploymenet` project.
+ 
+2. Go to `Preferences > Languages & Frameworks > PHP > Debug`. In this window there is an Xdebug section. Use these settings:
+    - Debug port: `9001`. (do NOT use the default, 9000)
```

Comment:
Not 9000 Because we have min.io right? maybe we could push it to 9999? Just to be sure we don't end using the next number for a service in the future?